src/config.ts

```diff
@@ -15,7 +16,7 @@ function getGlobalSetting (name: string) {
 	}
 	else if (hasDOM) {
 		return (
-			scriptElement.dataset[camelCaseName] ??
+			scriptElement!.dataset[camelCaseName] ??
```

Comment:
If I'm not mistaken, `scriptElement` will be defined here, but TS can't figure it out from the code, so we need to tell it we're sure with what we do.

Reply:
Maybe add a comment to that effect?

Reply:
Also, `scriptElement` won't necessarily be defined. `hasDOM` is a necessary, not a sufficient condition for it. It's also `null` when running code in ESM.

```suggestion
			scriptElement?.dataset[camelCaseName] ??
```